packages/turf-boolean-point-on-line/index.js

```diff
@@ -0,0 +1,65 @@
+var getCoords = require('@turf/invariant').getCoords;
+
+/**
+ * Returns true if a point is on a line. Accepts a optional parameter to ignore the start and end vertices of the linestring.
+ *
+ * @name booleanPointOnLine
+ * @param {Geometry|Feature<Point>} point GeoJSON Feature or Geometry
+ * @param {Geometry|Feature<LineString>} linestring GeoJSON Feature or Geometry
+ * @param {Boolean} [ignoreEndVertices=false] whether to ignore the start and end vertices.
+ * @returns {Boolean} true/false
+ * @example
+ * var pt = turf.point([0, 0]);
+ * var line = turf.lineString([[-1, -1],[1, 1],[1.5, 2.2]]);
+ * var isPointOnLine = turf.booleanPointOnLine(pt, line);
+ * //=true
+ */
+module.exports = function (point, linestring, ignoreEndVertices) {
+    var pointCoords = getCoords(point);
+    var lineCoords = getCoords(linestring);
+    for (var i = 0; i < lineCoords.length - 1; i++) {
```

Comment:
The only thing I find tricky about the invariant type modules is a lack of flow control, but in a lot of cases they are super-handy!